fenicsxconcrete/boundary_conditions/bcs.py

```diff
@@ -71,46 +80,54 @@ def add_dirichlet_bc(
             A hint which method should be used to locate the dofs.
             Choice: 'topological' or 'geometrical'.
         entity_dim : optional, int
-            The entity dimension in case `method=topological`.
+            The dimension of the entities to be located topologically.
+            Note that `entity_dim` is required if `sub` is not None and
+            `method=geometrical`.
         """
-        if boundary is None:
-            assert isinstance(value, dolfinx.fem.DirichletBCMetaClass)
+        if isinstance(value, dolfinx.fem.DirichletBCMetaClass):
             self._bcs.append(value)
         else:
             assert method in ("topological", "geometrical")
             V = self.V.sub(sub) if sub is not None else self.V
 
-            if method == "topological":
+            # dolfinx.fem.locate_dofs_geometrical(Vsub, plane_at) raises
+            # RuntimeError
+            topological = method == "topological" or sub is not None
```

Comment:
that's what the above comment was for. The reason why topological is true `if sub is not None` is that using `method=geometrical` raises a RuntimeError in this case.

Reply:
well, this might be my lack of understanding of how dolfinx works, but this comment did not help me to understand, I did not even see the direct connection. but I am not familiar with dolphinx, if you say that other people will understand this, then leave it.